application/controllers/Qso.php

```diff
@@ -119,6 +120,14 @@ public function index()
 	 * This is used for contest-logging and the ajax-call
 	 */
 	public function saveqso() {
+		// search qslmsg by default for this station (when contest or SFL) overload qslmsg post value if empty //
+		if (trim($this->input->post('qslmsg'))=='') {
+			$this->load->model('stations');
+			if (!(intval($this->input->post('station_profile'))>0)) {
+				$_POST['station_profile'] = $this->stations->find_active();
+			}
+			$_POST['qslmsg'] = $this->stations->get_station_info($_POST['station_profile'],'eqsl_defaultqslmsg', '');
```

Comment:
Why assigning values to $_POST?
$_POST is a var which contains the posted things. it's there to READ, not to transport data within the application.

Reply:
Further: Please check if station_id belongs to user! Otherwise there are security-risks. (Yes, i saw that you implemented it in the redundant function "get_station_info" at Stations-Model - but no need to implement redundant things here)

Reply:
Hello, 
1/ i use $_POST[] because : 
- $_POST['station_profile'] : not exist when a contest call saveqso() ; the station_profile is only defined in function create_qso() on logbook_model ; first time i would to add my test in this function but if user must an empty value qslmsg, i can not defined if a real empty value or it the qso create by SFL (SFL don't send  station_profile too ; even if we add a new information in ajax call in function js-save-to-lol(), like "isSFLQso" ?)
- $_POST['qslmsg'] : because it's not send by contest and SFL QSO save

and fonction logbook_model->create_qso(), use only information form POST, how can i send information to create_qso(), if i dont overload POST :(

2/ ok i can use check_station_is_accessible()

Reply:
you're right.
what about another way: prepopulate the eQSL-Msg-Field at the View upon loading the Form? I see more advantages here:
- User can change if he wants to type sth. special in it
- no need to set $_POST

Reply:
This is what is done in the context of QSOs (live/post), but in the case of a contest or SFL, the field does not exist...
I can add it by hiding it for contests, but not for SFL...
ok I have an idea, I will submit it